go_modules/lib/dependabot/go_modules/requirement.rb

```diff
@@ -132,13 +130,6 @@ def ruby_range(req_string)
         "~> #{parts.join('.')}"
       end
 
-      # NOTE: Dep's caret notation implementation doesn't distinguish between
```

Comment:
I removed this entire function because as best I can tell `go mod` has no special treatment for `^`...

Reply:
I think you're right, but we'll also need to remove the calling code and the tests in `go_modules/spec/dependabot/go_modules/requirement_spec.rb` that fail because of this removal

Reply:
So I started looking at this, and it looks like there's a _lot_ of tests in `go_modules/spec/dependabot/go_modules/requirement_spec.rb` that are simply inaccurate... Since `go mod` uses [MVS](https://golang.org/ref/mod#minimal-version-selection), most of the more traditional requirements ideas like ranges, wildcards, etc simply doesn't apply (IIUC).

So should the following be removed from the tests?
* `!`
* `^`
* `>`
* `*`
* `||`
* `,` (comma, separated list of versions)

Any of the above should result in a parsing error IMO, as they are not valid for `go.mod` syntax. 

Also, I'm confused why it looks like this is re-duplicating in ruby the `go mod` syntax checker? Why not just run `go mod tidy` and if it complains about syntax, then bomb out? 

Given the above, should all these helpers be removed?
* https://github.com/dependabot/dependabot-core/blob/80acab859af6322f6f160c1d30bc7e1777836a5e/go_modules/lib/dependabot/go_modules/requirement.rb#L63
* https://github.com/dependabot/dependabot-core/blob/80acab859af6322f6f160c1d30bc7e1777836a5e/go_modules/lib/dependabot/go_modules/requirement.rb#L67-L69
* https://github.com/dependabot/dependabot-core/blob/80acab859af6322f6f160c1d30bc7e1777836a5e/go_modules/lib/dependabot/go_modules/requirement.rb#L75-L119

I'm sure there's some stuff I'm ignorant of, so if there's a good reason to leave this all in, please educate me.

Reply:
Yeah, I noticed it as well, tbh I think you might be right and these all stem from the original `dep` implementation and are redundant for `go mod`, but just haven't bitten us because the regexes simply never match in real world scenario's.

cc @hmarr, it looks like you were the last person to really touch this code, could you confirm that we can nuke all those conversions?

Reply:
Nudge @hmarr any feedback? 

I'd like to push this across the line.

Reply:
It [looks like](https://github.com/dependabot/dependabot-core/commit/a23c650564ce168128f36acd6cf990cf97bbcf66) I did indeed just copy this file directly across from the dep implementation, which means it won't be correct for Go modules.

I'm definitely 👍  on making this code reflect how Go modules actually work. We might need to double check this won't break ignore conditions (which you can add to a config file, and also get generated when you say e.g. `@dependabot ignore this minor version`).

Reply:
> We might need to double check this won't break ignore conditions (which you can add to a config file, and also get generated when you say e.g. @dependabot ignore this minor version).

@jurre do you know if there are tests already covering this? I wasn't sure if there was a higher-level test spec that contains generic scenarios that all package managers need to satisfy, or if it needs to be done on a per-implementation level...

Reply:
nudge @jurre ☝️